tests/zfs-tests/tests/functional/block_cloning/block_cloning.kshlib

```diff
@@ -44,10 +44,14 @@ function have_same_content
 #
 function get_same_blocks
 {
+    KEY=$5
+    if [ ${#KEY} -gt 0 ]; then
+        KEY="--key=$KEY"
```

Comment:
You shouldn't need to do this. You don't need to decrypt the blocks to compare them.

Reply:
I thought this as well, until I got a lot permission denied for the checks. My first version of this contained raw key encrypted Datasets key files generated by openssl. And zdb can not decrypt raw key files.
Going to check it later and remove it if there is no error message.

Reply:
Ahh no, you're right actually. Don't need the key to see the block, but do need it to refer to it by name. No problem then!

Reply:
Instead of providing a key, I think we can resolve permission checks through the inode number:
```zdb -vvvv -d $dataset $(ls -i $file_path | awk '{print $1}')```